plugins/index.ts

```diff
@@ -0,0 +1,93 @@
+import { connect, MqttClient } from "mqtt";
+import { Plugin } from "@nuxt/types";
+import Vue from "vue";
+
+declare module "vue/types/vue" {
+    interface Vue {
+        /**
+         * Connects to Flespi client and fetch position data.
+         * @returns void
+         */
+        $getPositionData(): void
+    }
+}
+
+declare module "@nuxt/types" {
+    interface NuxtAppOptions {
+        $getPositionData(): void
+    }
+    interface Context {
+        $getPositionData(): void
+    }
+}
+
+declare module "vuex/types/index" {
+    interface Store<S> {
+        $getPositionData(): void
+    }
+}
+
+/**
+ * EventBus to transmit data to component.
+ */
+export const eventBus = new Vue(); // creating an event bus.
+
+function emitNewCoordinates (topic: string, content: string): void {
+    let longitude: number = 0;
+    let latitude: number = 0;
+
+    const splitTopic: string[] = topic.split("/");
+    const informationName: string = splitTopic[splitTopic.length - 1];
+
+    if (informationName === "position.longitude") {
+        longitude = Number(content);
+    } else if (informationName === "position.latitude") {
+        latitude = Number(content);
+    }
+
+    eventBus.$emit("newCoordinates", [longitude, latitude]);
+}
+
+const indexPlugin: Plugin = (context, inject) => {
```

Comment:
The file is named index.ts as it is the ts file related to index.vue

Reply:
That doesn't really make any sense, because if it's reused somewhere else, it is not anymore related to `index`. The reason why that component is called `index`, is because that is needed for Nuxt.js to understand that this page should be shown at the root of the website